presto-kudu/src/main/java/com/facebook/presto/kudu/KuduUtil.java

```
@@ -0,0 +1,99 @@
+            log.warn("----------current user: " + UserGroupInformation.getCurrentUser() + "----------");
+            log.warn("----------login user: " + UserGroupInformation.getLoginUser() + "----------");
+        }
+        catch (Exception e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    static KuduClient getKuduKerberosClient(KuduClientConfig config)
+    {
+        KuduClient client = null;
+        try {
+            reTryKerberos(true);
+            client = UserGroupInformation.getLoginUser().doAs(
+                    (PrivilegedExceptionAction<KuduClient>) () -> getKuduClient(config));
+        }
+        catch (Exception e) {
+            log.error(e.getMessage());
+        }
+        return client;
+    }
+
+    static KuduClient getKuduClient(KuduClientConfig config)
+    {
+        KuduClient.KuduClientBuilder builder = new KuduClient.KuduClientBuilder(config.getMasterAddresses());
+        builder.defaultAdminOperationTimeoutMs(config.getDefaultAdminOperationTimeout().toMillis());
+        builder.defaultOperationTimeoutMs(config.getDefaultOperationTimeout().toMillis());
+        builder.defaultSocketReadTimeoutMs(config.getDefaultSocketReadTimeout().toMillis());
+        if (config.isDisableStatistics()) {
+            builder.disableStatistics();
+        }
+        return builder.build();
+    }
+
+    static void reTryKerberos(boolean enabled)
+    {
+        if (enabled) {
+            log.warn("Try relogin kerberos at first!");
+            try {
+                if (UserGroupInformation.isLoginKeytabBased()) {
+                    UserGroupInformation.getLoginUser().reloginFromKeytab();
+                }
+                else if (UserGroupInformation.isLoginTicketBased()) {
+                    UserGroupInformation.getLoginUser().reloginFromTicketCache();
+                }
+            }
+            catch (IOException e) {
+                log.error("Try relogin kerberos failed!");
```

Ditto about exceptions. Rethrow instead of swallowing.